### PR TITLE
chore(README): add annotations documentation to README.md

### PR DESCRIPTION
**Pull request by sfinnman**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -60,7 +60,65 @@ See [schema.sql](./iamspanner/schema.sql).
...
+
+      option (einride.iam.v1.method_authorization) = {
+        permission: "namespace.entity.method"
+        before: {
+          expression: "test(caller, request.entity)" // iamcel expression
+          description: "The caller must have method permission against the entity"
+        }
+      };
+
+      /* ... */
```
Comment (ericwenn) [inline README.md:90]: IMO drop these comments

File: README.md [hunk 2]
```diff
@@ -60,7 +60,65 @@ See [schema.sql](./iamspanner/schema.sql).
...
+
+import "einride/iam/v1/annotations.proto";
+
+/* ... */
+
+service YourService {
+  /* ... */
+
+  rpc YourMethod(YourMethodRequest)
+    returns YourMethodResponse {
```
Comment (ericwenn) [inline README.md:79]: nit:
```suggestion
  rpc YourMethod(YourMethodRequest) returns YourMethodResponse {
```

File: README.md (hunk 1 above)
Comment (sfinnman) [reply README.md:90]: Added them for consistency, since the above examples included such comments. I'll drop them.

File: README.md [hunk 3]
```diff
@@ -60,7 +60,65 @@ See [schema.sql](./iamspanner/schema.sql).
...
+      /* ... */
+    };
+}
+
+/* ... */
+
+message YourMethodRequest {
+  string entity = 1 [
+    (google.api.resource_reference) = {
+      type: "www.example.com/Entity"
```
Comment (ericwenn) [inline README.md:99]: supernit:
```suggestion
      type: "example.com/Entity"
```

**Review by ericwenn — APPROVED** [review]
LGTM. Minor nits, but overall :star:

File: README.md (hunk 2 above)
Comment (ericwenn) [reply README.md:79]: yes please!

File: README.md (hunk 1 above)
Comment (ericwenn) [reply README.md:90]: Thanks!

**Review by sharkema — APPROVED** [review]

**Pull request merged**